gui/src/main.rs

```diff
@@ -30,6 +31,7 @@ pub struct GuiConfig {
     pub selected_http_method: HttpMethod,
     pub url: String,
     pub body_str: String,
+    pub headers: Rc<RefCell<Vec<(bool, String, String)>>>,
```

Comment:
Some fun interior mutability stuff, needed so that multiple owners can have shared access to this memory.

https://doc.rust-lang.org/book/ch15-05-interior-mutability.html

Reply:
I might be missing something stupid but wouldn't we want this to be a HashMap instead of a Vector to ensure we don't set the same header key twice?

Reply:
doing some more digging it seems like it might not matter, HashMap is constant time lookup but at the cost of more memory, vector is linear time but for small amounts of items Vector performs better? https://www.reddit.com/r/rust/comments/7mqwjn/hashmapstringt_vs_vecstringt/

Reply:
No, I was missing something...I totally wasn't thinking of someone adding a duplicate header hehe. That is an interesting point of the vector being faster at lower item count. I guess since we'll only have like 2 dozen headers at the very most we should stick with this? But can definitely enhance by trying to remove duplicate headers and only take the first encountered instance of a header value or something.